backport: secureboot support

sacnaik opened the pull request:
Backporting PR#12692 to the 202205 branch.

   The diffs of PR#12692 do not cleanly get applied to the 202205 branch. Hence patch #12692 was slightly modified for the 202205 branch.

<!--
     Please make sure you've read and understood our contributing guidelines:
     https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

     ** Make sure all your commits include a signature generated with `git commit -s` **

     If this is a bug fix, make sure your description includes "fixes #xxxx", or
     "closes #xxxx" or "resolves #xxxx"

     Please provide the following information:
-->

#### Why I did it
To support UEFI secure boot on the 202205 branch

#### How I did it
The feature is supported at the master branch see #12692. 
Backported #12692 from master to 202205 branch

#### How to verify it
Booted on UEFI secure boot-enabled hardware. 




#### Which release branch to backport (provide reason below if selected)

<!--
- Note we only backport fixes to a release branch, *not* features!
- Please also provide a reason for the backporting below.
- e.g.
- [x] 202006
-->

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### Description for the changelog
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->
Refer HLD: https://github.com/sonic-net/SONiC/pull/1028

#### Ensure to add label/tag for the feature raised. example - PR#2174 under sonic-utilities repo. where, Generic Config and Update feature has been labelled as GCU.

#### Link to config_db schema for YANG module changes
<!--
Provide a link to config_db schema for the table for which YANG model
is defined
Link should point to correct section on https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md
-->

#### A picture of a cute animal (not mandatory but encouraged)

rlhui commented:
@davidpil2002 , would you please review this one for 202205? thanks

ITJamie commented on installer/x86_64/install.sh:
```diff
@@ -569,7 +622,15 @@ if [ "$install_env" = "onie" ]; then
 
     if [ "$firmware" = "uefi" ] ; then
-        demo_install_uefi_grub "$demo_mnt" "$blk_dev"
+        secure_boot_state=$(mokutil --sb-state)
```
please note. I have had an issue with the latest master branch when i built for broadcom. it failed at this line as it could not find mokutil to check secureboot state
https://github.com/sonic-net/sonic-buildimage/issues/14316

sacnaik replied:
Are you using it on 202205? Looks like https://github.com/sonic-net/sonic-buildimage/issues/14316 is still open.
Once the issue is resolved I ll cherrypick them too. Thanks for letting me know.

ITJamie replied:
I was using it on master. 
Seems my onie env doesnt have the command and the hardware model hasnt had an official onie update. 

Would it make sense to have it use the command from the mounted image instead of from the onie env?

davidpil2002 replied:
I already suggested a solution to @ITJamie , Im pending to testing it with a device with old onie version.
I will add you as a reviewer as well.

sacnaik replied:
Hi David 
@ITJamie mentioned he will not use 202205. I can cherry-pick the specific fix later.
Can you review this PR?

davidpil2002 replied:
sure, you have differnt options:
1. install an update ONIE vertion than 2021.11
2. install manually in ONIE the mokutil command.
3. build the image including this fix: https://github.com/sonic-net/sonic-buildimage/pull/14429
Pls let me know if it solve the problem, thanks

kuanyu99 replied:
Thanks David!

sacnaik replied:
Backported this change.

sacnaik commented:
@davidpil2002 
1. ARM support is not needed at this time on 202205 branches and once you fully test I can cherry-pick that or move to newer branch
2. scripts/signing_secure_boot_prod.sh - I removed it
3. Added #14429 to the changes

davidpil2002 commented:
looks good to me, 
there is a small PR that will be publish soon with some small fixes, I think the best its to create this PR with the fixes by the day 1.
I approved this PR, but pls add the PR with the fixes.
@DavidZagury pls can you write in the comment the PR with your fixes so all the branches will be aligned?

davidpil2002 commented:
> @davidpil2002
> 
> 1. ARM support is not needed at this time on 202205 branches and once you fully test I can cherry-pick that or move to newer branch
> 2. scripts/signing_secure_boot_prod.sh - I removed it
> 3. Added [Secure boot fix instalation with devices that used ONIE version older than 2021.11 #14429](https://github.com/sonic-net/sonic-buildimage/pull/14429) to the changes

sorry for the inconvinence, but we decided to remove the `mokutil` dependency by using `efivar` tool. 
any chances you can cherry pick this new fix, instead the previous one.
new fix:
https://github.com/sonic-net/sonic-buildimage/pull/14589

You are welcome to review it as well

sacnaik commented:
Backported  https://github.com/sonic-net/sonic-buildimage/pull/14589 as well

yxieca reviewed (changes requested):
This change is not an approved feature for 202205 branch. Please request back porting to 202211 branch instead.

abdosi commented:
Already Backport msft repo 202205 branch.

gechiang commented:
Added the label "Icluded in Chassis for 202205 branch" label to keep the consistency where this PR was already backported by Abhishek separately even though this PR got closed.
